CRM/Event/Form/Task/Batch.php

```diff
@@ -268,7 +268,7 @@ public static function updatePendingOnlineContribution($participantId, $statusId
     $contributionId = CRM_Contribute_BAO_Contribution::checkOnlinePendingContribution($participantId,
       'Event'
     );
-    if (!$contributionId) {
+    if (!$contributionId || !$participantId) {
```

Comment:
I moved this up from lower down. It's actually impossible for it to be false but I thought I'd clarify that next round of cleanup by using type checking